src/sage/cpython/dict_del_by_value.pyx

```diff
@@ -48,7 +48,7 @@ cdef extern from "dict_internal.h":
 
 
 # dk_lookup was removed in python 3.11
-DEF HAS_DK_LOOKUP = PY_VERSION_HEX < 0x30b0000
+DEF HAS_DK_LOOKUP = False
```

Comment:
This is precisely what I had in mind.

Reply:
Nice. In this case, this PR is actually ready to go.